test/groovy/com/sap/piper/integration/TransportManagementServiceTest.groovy

```diff
@@ -178,4 +235,67 @@ class TransportManagementServiceTest extends BasePiperTest {
         assertThat(loggingRule.log, containsString("[TransportManagementService] Node upload successful."))
     }
 
+    @Test
+    void uploadFileToNode__failure() {
+        Map requestParams
+        helper.registerAllowedMethod('httpRequest', [Map.class], { m ->
+            requestParams = m
+            return [content: '{ "errorType": "TsInternalServerErrorException", "message": "The application has encountered an unexpected error." }', status: 400]
+        })
+        
+        def url = 'http://dummy.com/oauth'
+        def token = 'myToken'
+        def nodeName = 'myNode'
+        def fileId = 1234
+        def description = "My description."
+        def namedUser = 'myUser'
+        
+        def tms = new TransportManagementService(nullScript, [:])
+        def bodyRegEx = /^\{\s+"nodeName":\s+"myNode",\s+"contentType":\s+"MTA",\s+"description":\s+"My\s+description.",\s+"storageType":\s+"FILE",\s+"namedUser":\s+"myUser",\s+"entries":\s+\[\s+\{\s+"uri":\s+1234\s+}\s+]\s+}$/
+        def errorCaught = false
+        try {
+            tms.uploadFileToNode(url, token, nodeName, fileId, description, namedUser)
+        } catch (e) {
+            errorCaught = true
+            assertThat(e.getMessage(), is("Node upload failed. Consider re-running in verbose mode in order to get more details."))
```

Comment:
Strange pattern for checking for an exception. See above (`ExpectedException`, loggingRule.expect(./.)`